thinc/layers/with_reshape.py

```diff
@@ -1,15 +1,18 @@
-from typing import Tuple, Callable, Optional, cast
+from typing import Tuple, Callable, Optional, cast, TypeVar
```

Comment:
Here again I would appreciate a careful review.

Reply:
Looks right to me.